packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/launchpad/launch_coin.ts

```diff
@@ -64,19 +126,24 @@ export const launchCoin = async (
     }
 
     if (
-      typeof initialBuyAmountAudio !== 'number' ||
-      initialBuyAmountAudio <= 0 ||
-      isNaN(initialBuyAmountAudio)
+      initialBuyAmountSol !== undefined &&
+      (typeof initialBuyAmountSol !== 'number' ||
+        initialBuyAmountSol <= 0 ||
+        isNaN(initialBuyAmountSol))
```

Comment:
`typeof` check in conjunction with `isNaN` feels redundant? any reason for both?